### PR TITLE
Pull request for the re-form branch

### PR DESCRIPTION
**Pull request by Datagirlcmr**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def user_params
+    params.require(:user).permit(:username, :password, :email)
+  end
```
Comment (Redvanisation) [inline app/controllers/users_controller.rb:35]: This method has to be private

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <% end %>
+    </ul>
+  </div>
+<% end %>
+<%= form_for @user do |f| %>
+  <%= label_tag(:username) %>
+  <%= f.text_field :username %>
+  <%= label_tag(:password) %>
+  <%= f.password_field :password %>
+  <%= label_tag(:email) %>
```
Comment (Redvanisation) [inline app/views/users/new.html.erb]: Instead of using `label_tag` in a `form_for` you could use `f.label` making use of the form's object.

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Very nice work on this project, however, the following issues need to be fixed before moving on ⬇️** 

### Users/views - new:
- ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) There must be a commented out form which:
- [ ]  uses users_path helper for action
- [ ]  uses post for method
- [ ]  uses "UTF-8" for ccept-charset
- [ ]  uses user[<field_name>] for input's name
- [ ]  uses form_authenticity_token

- ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) There must be a commented out `form_tag` that: 
* [ ] uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`

### Users/views - edit:
- The same thing as the **new** forms above, you have to put the commented out forms for the edit view 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).

- Please see my comments below and fix the related issues.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

**great work so far, you are almost there, you just left out one requirement from the previous review ⬇️:**

> 
> ### Users/views - edit:
> The same thing as the new forms above, you have to put the commented out forms for the edit view
> (screenshot from Odin).

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+<%# </form> %>
+
+<%# <%= form_tag("/users", method: "post") do %>
+<%# <%= label_tag(:username) %>
+<%# <%= text_field_tag("user[:username]") %>
+<%# <%= label_tag(:password) %>
+<%# <%= password_field_tag("user[:password]") %>
+<%# <%= label_tag(:email) %>
+<%# <%= email_field_tag("user[:email]") %>
+<%# <%= submit_tag('submit')%>
```
Comment (samgaco) [inline app/views/users/edit.html.erb:19]: :medal_military:

**Review by samgaco — APPROVED** [review]
The previous requirement is covered hence the project is approved :white_check_mark:  You can move on! :muscle: 

---------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

File: app/views/users/edit.html.erb (hunk 3 above)
Comment (Datagirlcmr) [reply app/views/users/edit.html.erb:19]: Thank You. Can you write your full names so we use in submitting the project completion form

**Pull request merged**